ui/src/app/pages/data/criteria-Search.tsx

```diff
@@ -0,0 +1,234 @@
+import {ListSearchV2} from 'app/cohort-search/list-search-v2/list-search-v2.component';
+import {Selection} from 'app/cohort-search/selection-list/selection-list.component';
+import {CriteriaTree} from 'app/cohort-search/tree/tree.component';
+import colors, {colorWithWhiteness} from 'app/styles/colors';
+import {reactStyles} from 'app/utils';
+import {
+  attributesSelectionStore,
+  currentCohortCriteriaStore,
+  currentConceptStore
+} from 'app/utils/navigation';
+import {Criteria, DomainType} from 'generated/fetch';
+import * as fp from 'lodash/fp';
+import {Growl} from 'primereact/growl';
+import * as React from 'react';
+import {Subscription} from 'rxjs/Subscription';
+
+const styles = reactStyles({
+  growl: {
+    position: 'absolute',
+    right: '0',
+    top: 0
+  }
+});
+const css = `
+  .p-growl {
+    position: sticky;
+  }
+  .p-growl.p-growl-topright {
+    height: 1rem;
+    width: 6.4rem;
+    line-height: 0.7rem;
+  }
+  .p-growl .p-growl-item-container .p-growl-item .p-growl-image {
+    font-size: 1rem !important;
+    margin-top: 0.19rem
+  }
+  .p-growl-item-container:after {
+    content:"";
+    position: absolute;
+    left: 97.5%;
+    top: 0.1rem;
+    width: 0px;
+    height: 0px;
+    border-top: 0.5rem solid transparent;
+    border-left: 0.5rem solid ` + colorWithWhiteness(colors.success, 0.6) + `;
+    border-bottom: 0.5rem solid transparent;
+  }
+  .p-growl-item-container {
+    background-color: ` + colorWithWhiteness(colors.success, 0.6) + `!important;
+  }
+  .p-growl-item {
+    padding: 0rem !important;
+    background-color: ` + colorWithWhiteness(colors.success, 0.6) + `!important;
+    margin-left: 0.3rem;
+  }
+  .p-growl-message {
+    margin-left: 0.5em
+  }
+  .p-growl-details {
+    margin-top: 0.1rem;
+  }
+ `;
+
+interface Props {
+  cohortContext: any;
+  source: string;
+  showHierarchy?: Function;
+}
+
+interface State {
+  backMode: string;
+  autocompleteSelection: Criteria;
+  growlVisible: boolean;
+  groupSelections: Array<number>;
+  hierarchyNode: Criteria;
+  mode: string;
+  selections: Array<Selection>;
+  selectedCriteriaList: Array<any>;
+  selectedIds: Array<string>;
+  treeSearchTerms: string;
+  loadingSubtree: boolean;
+
+}
+export class CriteriaSearch extends React.Component<Props, State>  {
```

Comment:
nit: make the 's' lowercase in component filename

Reply:
Done